build.d: Change  working directory to the project root

MoonlightSentinel opened the pull request:
.. instead of `src`. This ensures that error messages, coverage data, ... will be reported relative to the root directory as expected by many IDE's.

MoonlightSentinel committed:
build.d: Change  working directory to the project root ...

.. instead of `src`. This ensures that error messages, coverage data, ...
will be reported relative to the root directory as expected by many IDE's.

Also adjusted the CodeCov script to deal with the new paths.

MoonlightSentinel commented:
This breaks CodeCov ...

wilzbach commented on .circleci/run.sh:
```diff
@@ -193,9 +193,9 @@ codecov()
     download "https://codecov.io/bash" "https://raw.githubusercontent.com/codecov/codecov-bash/master/codecov" "codecov.sh"
-    cd src # need to run from compilation folder for gcov to find sources
-    bash ../codecov.sh -p .. -x gcov-4.9 # must match g++ version (on CircleCI `g++` is 4.9 and `gcov` 4.6 :/)
+    bash codecov.sh -p . -s src -Z -x gcov-4.9 # must match g++ version (on CircleCI `g++` is 4.9 and `gcov` 4.6 :/)
```
We don't have C or C++ files anymore, so you could drop the gcov flag here.

wilzbach reviewed (commented):
In mars.d the coverage base directory is set. I would guess that this needs adjustment too.

MoonlightSentinel committed:
Improve CodeCov upload

Removes the now obsolete gcov plugin (no C++ sources remaining) and
fail if any error occurred in the CodeCov script

MoonlightSentinel commented:
Thanks, CodeCov works again

Geod24 commented:
Wild guess: the shenanigans in `mars.d` predate https://github.com/dlang/druntime/pull/1916 and could be replaced now.

Geod24 commented on .circleci/run.sh:
```diff
@@ -194,8 +194,7 @@ codecov()
     rm -rf test/runnable/extra-files
     download "https://codecov.io/bash" "https://raw.githubusercontent.com/codecov/codecov-bash/master/codecov" "codecov.sh"
-    cd src # need to run from compilation folder for gcov to find sources
-    bash ../codecov.sh -p .. -x gcov-4.9 # must match g++ version (on CircleCI `g++` is 4.9 and `gcov` 4.6 :/)
```
What's the story here ?

MoonlightSentinel replied:
Changing the working directory implies different paths for coverage files.
`dmd_expressionsem.lst` located in `src` becomes `src_dmd_expressionsem.d` in the root directory.

See https://github.com/dlang/dmd/pull/11534#discussion_r467500690 regarding `gcov`

MoonlightSentinel commented:
> Wild guess: the shenanigans in `mars.d` predate [dlang/druntime#1916](https://github.com/dlang/druntime/pull/1916) and could be replaced now.

Probably, but embedding `rt_options` might be better suited to keep the current behaviour?
